### PR TITLE
[BFD-760] Modify New Relic config in PipelineApplication

### PR DESCRIPTION
**Pull request by rtravitz**
### Change Details
This PR addresses [BFD-760](https://jira.cms.gov/browse/BFD-760). It changes how New Relic is configured to use the `AppConfiguration` class. This is happening [after a request here](https://github.com/CMSgov/beneficiary-fhir-data/pull/536#issuecomment-824342566).

### Acceptance Validation
- Configuration options still are picked up and work to send metrics to New Relic.

### External References
- [BFD-760](https://jira.cms.gov/browse/BFD-760)

### Security Implications
- [ ] new software dependencies
- [ ] altered security controls
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Comment by jzulim** [comment]
@rtravitz This looks legit, but almost impossible to test locally. Should we fire it off to TEST and wait 20 minutes to see if it 'works' - :/

**Comment by rtravitz** [comment]
No errors in the pipeline application after deploy. The metrics are flowing to New Relic. The only thing that appears to need fixing is the `appName` is getting set as null. May need to change the name of the environment variable.

**Comment by rtravitz** [comment]
It was out of line with the name of the environment variable injected by the `ops` code at startup. It's fixed now, so this is good for a full review!

**Comment by cbrunefearless** [comment]
> It was out of line with the name of the environment variable injected by the `ops` code at startup. It's fixed now, so this is good for a full review!

Tested in Test?

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java [hunk 1]
```diff
@@ -127,6 +169,8 @@ public String toString() {
     builder.append(extractionOptions);
     builder.append(", loadOptions=");
     builder.append(loadOptions);
+    builder.append(", metricOptinos=");
```
Comment (karlmdavis) [inline apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:172]:
```suggestion
    builder.append(", metricOptions=");
```

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/MetricOptions.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
+package gov.cms.bfd.pipeline.app;
+
+import java.io.Serializable;
+
+/** Models the user-configurable options for sending telemetry to New Relic. */
+public final class MetricOptions implements Serializable {
```
Comment (karlmdavis) [inline apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/MetricOptions.java:6]: You'll have a warning here that it needs a default version blah blah blah field, I think?

**Review by karlmdavis — DISMISSED** [review]
One minor fix and one minor note. Good to go, otherwise.

**Comment by rtravitz** [comment]
> > It was out of line with the name of the environment variable injected by the `ops` code at startup. It's fixed now, so this is good for a full review!
> 
> Tested in Test?

Just changed the name of the app name env var to match up. Not needed in this case!

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/MetricOptions.java (hunk 2 above)
Comment (rtravitz) [reply apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/MetricOptions.java:6]: Will look into it

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java (hunk 1 above)
Comment (rtravitz) [reply apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:172]: Nice catch

**Review by karlmdavis — APPROVED** [review]
🚀

**Review by jzulim — APPROVED** [review]
Nice having config / vars in one place. Thanks!

**Pull request merged**